File: backend/mmr/src/main/java/cit/edu/mmr/dto/ReportDTO.java
package cit.edu.mmr.dto;

import cit.edu.mmr.entity.UserEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ReportDTO {
    private long id;
    private long reportedID;
    private String itemType;
    private UserDTO reporter;  // Changed from UserEntity to UserDTO
    private String status;
    private Date date;
}

File: frontend-mobile/Memoire/app/src/main/java/com/example/memoire/models/ReportDTO.kt
package com.example.memoire.models

import java.util.Date

data class ReportDTO(
    val id: Long,
    val reportedID: Long,
    val itemType: String,
    val reporter: UserDTO, // Assuming UserDTO is already defined
    val status: String,
    val date: Date
)

data class ReportRequest(
    val reporterId: Long,
    val reportedID: Long,
    val itemType: String,
    val status: String
)